src/liboslexec/backendllvm.cpp

```diff
@@ -949,13 +949,13 @@ BackendLLVM::userdata_initialized_ref (int userdata_index)
 
 llvm::Value *
 BackendLLVM::llvm_call_function (const char *name, 
-                                      const Symbol **symargs, int nargs,
-                                      bool deriv_ptrs)
+                                 OIIO::cspan<const Symbol *> args,
+                                 bool deriv_ptrs)
 {
-    std::vector<llvm::Value *> valargs;
-    valargs.resize ((size_t)nargs);
-    for (int i = 0;  i < nargs;  ++i) {
-        const Symbol &s = *(symargs[i]);
+    llvm::Value *valargs[16];
+    DASSERT(args.size() < int(sizeof(valargs) / sizeof(valargs[0])));
```

Comment:
I'm not crazy about this. Looks like  bug lurking if there's ever a function with > 16 args.
Let's leave the vector in this one case. Good candidate someday for a short_vector.

Reply:
Just to be clear - this would only trigger for a function handled by `gen_generic`. It seems a bit unlikely to have more than 16 arguments for a function that didn't need special handling at all.

But I'm happy to put in an overflow case that goes through `std::vector`.